core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/utils/RetryableStageHelper2.java

```diff
@@ -162,8 +162,10 @@ public SdkException retryPolicyDisallowedRetryException() {
                                   .message("Request attempt " + (i + 1) + " failure: " + exceptionMessageHistory.get(i))
                                   .writableStackTrace(false)
                                   .build();
+
             lastException.addSuppressed(pastException);
         }
+        lastException.setAttempts(retriesAttemptedSoFar() + 1);
```

Comment:
We should probably use `toBuilder().numAttempts(x).build()`

Reply:
I wanted to do that initially, but toBuilder will cause the re-consturction of the exception message and will result in a duplication of the attempt count (Attempt: N) (Attempt N).

Discussed this with @millems and @dagnir  and since we are already mutating the exception that mutating is reasonable.